mypyc/lib-rt/CPy.h

```diff
@@ -930,6 +930,42 @@ static PyObject *CPyDict_FromAny(PyObject *obj) {
     }
 }
 
+static PyObject *CPyStr_GetItem(PyObject *str, CPyTagged index) {
+    if (PyUnicode_Check(str) && PyUnicode_READY(str) != -1) {
+        if (CPyTagged_CheckShort(index)) {
+            Py_ssize_t n = CPyTagged_ShortAsSsize_t(index);
+            Py_ssize_t size = PyUnicode_GET_LENGTH(str);
+            if ((n >= 0 && n >= size) || (n < 0 && n + size < 0)) {
+                PyErr_SetString(PyExc_IndexError, "string index out of range");
+                return NULL;
+            }
+            if (n < 0)
+                n += size;
+            enum PyUnicode_Kind kind = (enum PyUnicode_Kind)PyUnicode_KIND(str);
+            void *data = PyUnicode_DATA(str);
+            Py_UCS4 ch = PyUnicode_READ(kind, data, n);
+            PyObject *unicode = PyUnicode_New(1, ch);
+            if (unicode == NULL)
+                return NULL;
+
+            assert(PyUnicode_KIND(unicode) != PyUnicode_1BYTE_KIND);
```

Comment:
This seems wrong to me. Why can't it be 1BYTE? Looking at the code in `unicodeobject.c` it seems like it should be able to be?